audinterface/core/utils.py

```diff
@@ -463,16 +463,24 @@ def duration_in_seconds(duration, sampling_rate):
     ):
         # sequence of duration entries
         durations = [
-            duration_in_seconds(duration, sampling_rate)
+            to_timedelta(duration, sampling_rate)
             for duration in durations
         ]
     else:
         # single duration entry
-        durations = duration_in_seconds(durations, sampling_rate)
-
-    durations = pd.to_timedelta(durations, unit='s')
-
-    if isinstance(durations, pd.TimedeltaIndex):
-        durations = list(durations)
+        if not isinstance(durations, pd.Timedelta):
+            durations = duration_in_seconds(durations, sampling_rate)
+            durations = pd.to_timedelta(durations, unit='s')
+            # Limit precision to 6 digits
+            # to mimick previous behavior of pandas
+            # compare https://github.com/audeering/audinterface/issues/113
+            precision = 6
+            durations_str = str(durations)
+            durations_str_parts = durations_str.split('.')
+            if len(durations_str_parts) > 1:  # check if we have digits at all
+                num_digits = len(durations_str_parts[-1])
+                if num_digits > precision:
+                    durations_str = durations_str[:precision - num_digits]
+                    durations = pd.Timedelta(durations_str)
```

Comment:
The tests are currently failing as we compare some results against indices that we generate with `audformat.segmented_index()` which does not have the limiting to 6 digits included yet (see also https://github.com/audeering/audformat/pull/377).

@jwagner: The question is also if we still want to have it or not. It would make the `end` times compatible to what we will get for most of our databases, but as we might have published databases in the mean time using a newer version of `pandas` we will not be able to return the "correct" end time for all databases.
It's basically that `pandas` had a bug for several years in how they rounded to 6 digits and now this is fixed and creates a big mess.

Reply:
NOTE: this part of the code is only needed to ensure executing `process_index()` on a filewise index will return the same result as calling `get(as_segmented=True, allow_nat=False)` in `audformat`.

Reply:
On the other hand if we decide to ensure the old behavior we don't need to pass on the `start` and `end` values inside the process function as I do at the moment and can again rely on the calculation of the duration by the processed samples, which seems in general a good approach.

We could then move mimicking the old rounding error behavior of `pandas` to `audformat` and reuse that function here instead of implementing it again.

As I said the only downside (besides that the duration value is wrong by around 1 microsecond) might be that we had published a few databases in the mean time that have stored duration values using the new `pandas` behavior.

Reply:
It might also be not a `pandas` bug as I was getting the same behavior for `pd.to_timedelta()` for pandas v1.2.0, but rounding errors we introduce when converting to seconds and back from the values we store in the dependency CSV table. But I have no idea why this problem did not happen before.

Reply:
Would it help if we add a `precision` argument? But I guess we would need to add this in several places like `audformat.Database.load()`, `audformat.segmented_index()`, `audformat.utils.to_segmented_index()`, etc.

Reply:
It might be that this issue is caused by https://github.com/audeering/audformat/issues/379 and we don't need to change anything in `audinterface` afterthis is fixed in `audformat`.